sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSonarWayProfile.java

```diff
@@ -25,37 +25,27 @@
 import java.util.Set;
 
 import org.sonar.api.SonarRuntime;
-import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition;
-import org.sonar.api.utils.Version;
 import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
-import org.sonarsource.analyzer.commons.BuiltInQualityProfileJsonLoader;
+import org.sonarsource.dotnet.shared.plugins.AbstractSonarWayProfile;
 
 import javax.annotation.Nullable;
 
-public class CSharpSonarWayProfile implements BuiltInQualityProfilesDefinition {
+public class CSharpSonarWayProfile extends AbstractSonarWayProfile {
   private static final Logger LOG = Loggers.get(CSharpSonarWayProfile.class);
-  private final boolean supportsSecurityHotspots;
-  private static final Version SQ_7_3 = Version.create(7, 3);
 
   CSharpSonarWayProfile(@Nullable SonarRuntime sonarRuntime) {
-    this.supportsSecurityHotspots = sonarRuntime != null && sonarRuntime.getApiVersion().isGreaterThanOrEqual(SQ_7_3);
+    super(sonarRuntime, CSharpPlugin.LANGUAGE_KEY, CSharpPlugin.PLUGIN_KEY, CSharpPlugin.REPOSITORY_KEY);
   }
 
   @Override
-  public void define(Context context) {
```

Comment:
Extracted the common code that determines which sonar way profile definition to load depending on the SQ version